### PR TITLE
Implement filtering by topics for events

### PR DESCRIPTION
**Pull request by nlordell**
This PR implements filtering event streams by topic.

### Test Plan

Unit test and example

### CODE REVIEW

**Review by e00E — APPROVED** [review]

File: examples/events.rs [hunk 1]
```diff
@@ -32,7 +32,9 @@ async fn run() {
         .instance
         .event::<_, (Address, Address, U256)>("Transfer")
         .expect("transfer event not found")
-        .stream();
+        .topic0(Topic::This(accounts[0])) // from
```
Comment (fleupold) [inline examples/events.rs:35]: Not the nicest API, but I assume we will have to wait until we do code-generation to have named arguments here...

**Review by fleupold — APPROVED** [review]
📽

File: examples/events.rs (hunk 1 above)
Comment (nlordell) [reply examples/events.rs:35]: Exactly, that's what I'm working on now.

Comment (nlordell) [reply examples/events.rs:35]: @e00E also suggested `all_events` stream an `enum Event { Transfer(...), Mint(...), etc. }`.

**Pull request merged**